src/category_theory/sites/surjective.lean

```diff
@@ -0,0 +1,159 @@
+/-
+Copyright (c) 2022 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import category_theory.sites.subsheaf
+
+/-!
+
+# Image sheaf
+
+## Main definitions
+
+- `is_surjective` : A morphism of presheaves valued in a concrete category is surjective if
+  every section in the target is locally in the set-theoretic image,
```

Comment:
I think this docstring (and maybe also some others below) should point out that this depends on a Grothendieck topology. Because `presheaf` doesn't necessarily imply that on its own.

Reply:
This definition does not seem to be standard. [The Stacks project](https://stacks.math.columbia.edu/tag/00V5) defines a surjective map of presheaves simply to mean it's surjective on every open, and shows being surjective is equivalent to being an epimorphism. It's stated for Set-valued presheaves, but I think it's generalizable to functor categories with a suitable concrete category as the target. This definition would not depend on the Grothendieck topology and should probably go into the functor category part of mathlib. (I think the definition of subpresheaf also need not depend on the G topology; they're just subfunctors and could use the dependent functor formulation that I proposed but is not yet in mathlib.) [For sheaves](https://stacks.math.columbia.edu/tag/00WL) you do need the "locally in the image" to characterize the epimorphisms. @jjaassoonn has done something about this [for monomorphisms](https://github.com/leanprover-community/mathlib/pull/15934/files#r958826181):
> Sheaf(C, X) morphism is mono iff it’s a mono on every open holds for every concrete category C that preserves pullback, this is in the other PR containing the sky-stalk adjunction.

Reply:
This should be standard when the target is a sheaf? I allowed presheaves so that I could claim that the sheafification morphism is surjective. 

I probably should make this clearer / change the name nonetheless.

Reply:
As for surjective natural transformation, we already have [`epi_iff_epi_app`](https://leanprover-community.github.io/mathlib_docs/category_theory/adjunction/evaluation.html#category_theory.nat_trans.epi_iff_epi_app) (for categories with certain products), and `epi_iff_surjective` for most concrete categories whose forgetful functor reflects epimorphisms, so I think using `[epi f]` is easy enough to use and there doesn't seem to be a need for a separate definition.

Reply:
Thanks for pointing out [epi_iff_epi_app](https://leanprover-community.github.io/mathlib_docs/category_theory/adjunction/evaluation.html#category_theory.nat_trans.epi_iff_epi_app). Yes in the context of sheaves your definition is standard, but since you are defining it for presheaves and using G toplogy in the definition, I think naming it `is_locally_surjective` is more appropriate.

Reply:
> Also, the current subpresheaf API also does not depend on a topology as well.

Yes indeed; it would be clearer if it's not in the grothendieck_topology namespace. I must have gotten some illusion from that namespace!

By the way do you plan to make the connection between your is_surjective and epimorphisms in the category of sheaves?

Reply:
@erdOne what's the status here? Can this discussion be resolved? Is the PR ready for review again?

Reply:
I renamed the predicate to `is_locally_surjective` and fixed the docs. 
I'll deal with the namespace issue of `subpresheaf` in a later PR.
`is_locally_surjective` iff `epi` will also be provided in a later PR.